src/main/java/bio/terra/workspace/service/workspace/flight/GenerateControlledResourceIdStep.java

```diff
@@ -0,0 +1,24 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class GenerateControlledResourceIdStep implements Step {
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    flightContext
+        .getWorkingMap()
+        .put(WorkspaceFlightMapKeys.CONTROLLED_RESOURCE_ID, UUID.randomUUID());
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext flightContext) throws InterruptedException {
```

Comment:
should this `undoStep` technically clear the entry from the map?

Reply:
That isn't the common practice.

Reply:
So only effects on the external environment then.

Reply:
Discussed offline, and this isn't necessary, as the sequence isn't restarted in the middle.